src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -98,23 +110,29 @@ public class GenerateProtoTask extends DefaultTask {
      *
      * Default: null
      */
+    @Nullable
+    @Optional
+    @Input
     GString path
 
     /**
      * If true, source information (comments, locations) will be included in the descriptor set.
      *
      * Default: false
      */
+    @Input
     boolean includeSourceInfo
 
     /**
      * If true, imports are included in the descriptor set, such that it is self-containing.
      *
      * Default: false
      */
+    @Input
     boolean includeImports
   }
 
+  @Nested
   final DescriptorSetOptions descriptorSetOptions = new DescriptorSetOptions()
```

Comment:
I suppose this is only an input when `generateDescriptorSet` is true.

Reply:
Good point.